src/listContext.js

```diff
@@ -31,27 +30,72 @@ const ListContextProvider = props => {
       .catch(function(error) {
         console.error('Error getting shopping list ', error);
       });
+    return tempArray;
   };
 
-  function isDuplicate(name) {
+  const isDuplicate = name => {
     let normalizedName = normalizeName(name);
     let normalizedList = shoppingList.map(item => normalizeName(item.name));
     const isDupe = normalizedList.includes(normalizedName);
     return isDupe;
-  }
+  };
+  const validToken = token => {
+    let filteredByToken = shoppingList.filter(item => item.token === token);
+    console.log('filtered by token', filteredByToken);
+    console.log('includes token? ', filteredByToken.length > 0);
+    return filteredByToken.length > 0;
+  };
 
-  function addItem(name, nextExpectedPurchase) {
+  const addItem = (name, nextExpectedPurchase) => {
     if (!isDuplicate(name)) {
       itemsRef.add({ name, token, nextExpectedPurchase });
       fetchList(token);
       setName('');
     }
-  }
+  };
+
+  const displayList = token => {
```

Comment:
@mikeramz86 @MonicaDJohnson @ajiles91 I moved the list rendering to listContext. It's almost unchanged but in a completely different spot so I just wanted to give heads up in case this gets merged soon.

Reply:
I see what you were going for here @prophen. It's a great idea to make rendering the list easier!

Though, this is something I would push back hard on in a company PR. Adding UI to the Context here tightly couples the list to _always_ being rendered this way. Perhaps we'd want list data and to show only a subset of that data. 

I think we want to separate data fetching/transformation and UI creation. A component consuming this context should have all it needs to render the list. If we want to reuse that particular UI, it should be its own component and passed the list data as a prop.

I would probably use `withFirebase` to get the firebase API directly and export that data out of context.